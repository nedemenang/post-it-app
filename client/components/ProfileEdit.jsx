import React, { Component } from 'react';
import ImagesUploader from 'react-images-uploader';
import 'react-images-uploader/styles.css';
import 'react-images-uploader/font.css';
import '../public/style.scss';
import TitleBar from './TitleBar';
import History from '../utils/history';
import { updateUserProfile } from '../actions/AppActions';

/**
 * @class ProfileEdit
 *
 * @extends {Component}
 */
class ProfileEdit extends Component {

/**
 * Call update user profile action
 *
 * @param {object} event event object
 *
 * @memberof ProfileEdit
 */

 /**
   * Set username state when user types
   *
   * @param {object} event event object
   *
   * @memberof ProfileEdit
   *
   * @returns {void} returns void
   */
  handleUserNameChange(event) {
    this.setState({ userName: event.target.value });
  }


  /**
   * Set phone number state when user selection changes
   *
   * @param {object} event event object
   *
   * @memberof MessageForm
   *
   * @returns {void} returns void
   */
  handlePhoneNumberChange(event) {
    this.setState({ phoneNumber: event.target.value });
  }


  /**
   * Saves the user information on click of button
   *
   * @param {object} event event object
   *
   * @memberof ProfileEdit
   *
   * @return {void} returns void
   */
  save(event) {
    event.preventDefault();
    const user = {
      userName: this.state.userName,
      phoneNo: this.state.phoneNumber,
      profilePic: this.state.file
    };
    updateUserProfile(user);
    this.setState({
      userName: '',
      phoneNumber: ''
    });
    setTimeout(() => {
      History.push('/');
    }, 2000);
  }


  /**
   * Creates an instance of ProfileEdit.
   *
   * @param {object} props props object
   *
   * @memberof ProfileEdit
   */
  constructor(props) {
    super(props);
    const str = location.href;
    const str2 = str.replace('#', ' ');
    this.state = {
      file: '',
      path: str2,
      userName: '',
      phoneNumber: ''
    };
    this.save = this.save.bind(this);
  }

  /**
   * Renders profile edit page
   *
   * @returns {JSX} profile edit page
   *
   * @memberof ProfileEdit
   */
  render() {
    return (
   <div>
     <TitleBar />
     <div className="form">
    <form className="login-form">
    <h3 style={{ marginTop: 40 }}>Profile Edit</h3>
      <input type="text"
      value={this.state.userName}
      id="userName"
      placeholder="username"
      onChange={ this.handleUserNameChange.bind(this) }/>
      <input type="text"
      id="phoneNumber"
      value={this.state.phoneNumber}
      placeholder="Phone number (+2348012345678)"
      onChange={ this.handlePhoneNumberChange.bind(this) }
      />
      <ImagesUploader
                url={`${this.state.path}/profilePictures`}
                optimisticPreviews
                multiple={false}
                onLoadEnd={(error, response) => {
                  this.setState({
                    file: response,
                  });
                }}
                label="Upload a profile picture"
                />
      <button className="button" onClick={this.save}>Save</button>
    </form>
  </div>
</div>
    );
  }

}

export default ProfileEdit;
