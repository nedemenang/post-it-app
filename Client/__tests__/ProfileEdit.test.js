import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import appAction from '../actions/AppActions';
import ProfileEdit from '../components/ProfileEdit.jsx';

describe('the profile Edit component', () => {
  let mountedComponent;

  const str = location.href;
  const str2 = str.replace('#', ' ');

  const loggedInUser = {
    displayName: 'nnamisco',
    email: 'nedemenang@gmail.com',
    id: 'TUsUaucS5nb5kuOP6SpvTUcX9vD3',
    isAuthenticated: true,
    phoneNo: null,
    profilePic: `${str2}static/files/blank-profile-pic.png`
  };

  const ProfileEditItem = () => {
    if (!mountedComponent) {
      mountedComponent = shallow(<ProfileEdit />);
    }
    return mountedComponent;
  };

  beforeEach(() => {
    mountedComponent = undefined;
  });

  it('should render', () => {
    const component = ProfileEditItem();
    expect(component).toBeDefined();
  });

  it('should match snapshot test', () => {
    const component = ProfileEditItem();
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });

  it('should update state on username field change', () => {
    const input = ProfileEditItem().find('#userName').first();
    input.simulate('change', { target: {
      value: 'Test Username'
    } });
    expect(ProfileEditItem().state().userName).toBe('Test Username');
  });

  it('should update state on phonenumber field change', () => {
    const input = ProfileEditItem().find('#phoneNumber').first();
    input.simulate('change', { target: {
      value: '+23470239403923'
    } });
    expect(ProfileEditItem().state().phoneNumber).toBe('+23470239403923');
  });
});
