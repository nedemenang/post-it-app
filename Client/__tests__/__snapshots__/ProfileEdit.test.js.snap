// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the profile Edit component should match snapshot test 1`] = `
<div>
  <TitleBar />
  <div
    className="form"
  >
    <form
      className="login-form"
    >
      <h3
        style={
          Object {
            "marginTop": 40,
          }
        }
      >
        Profile Edit
      </h3>
      <input
        id="userName"
        onChange={[Function]}
        placeholder="username"
        type="text"
        value=""
      />
      <input
        id="phoneNumber"
        onChange={[Function]}
        placeholder="Phone number (+2348012345678)"
        type="text"
        value=""
      />
      <ImagesUploader
        borderColor="#a9bac8"
        classNames={Object {}}
        classNamespace="iu-"
        color="#142434"
        dataName="imageFiles"
        disabledBorderColor="#bec3c7"
        disabledColor="#bec3c7"
        headers={Object {}}
        label="Upload a profile picture"
        multiple={false}
        notificationBgColor="rgba(0, 0, 0, 0.3)"
        notificationColor="#fafafa"
        onLoadEnd={[Function]}
        optimisticPreviews={true}
        styles={Object {}}
        url="about:blank/profilePictures"
      />
      <p
        className="success"
      />
      <p
        className="error"
      />
      <button
        className="button"
        onClick={[Function]}
      >
        Save
      </button>
    </form>
  </div>
</div>
`;
